File: problems/563/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ans = 0;
    int findTilt(TreeNode* root) {
        traverse(root);
        
        return ans;
    }
    
    int traverse(TreeNode* node) {
        if (!node) {
            return 0;
        }
        
        int left = traverse(node->left), right = traverse(node->right);
        ans += abs(left - right);
        
        return left + right + node->val;
    }
};

File: problems/563/solution.go
package main
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func findTilt(root *TreeNode) int {
    ans := 0
    
    traverse(root, &ans)
    
    return ans
}

func traverse(node *TreeNode, ans *int) int {
    if node == nil {
        return 0
    }
    
    left, right := traverse(node.Left, ans), traverse(node.Right, ans)
    
    *ans += abs(left - right)
    
    return left + right + node.Val
}

func abs(a int) int {
    if a < 0 { return a * -1 }
    return a
}